src/transformers/modeling_utils.py

```diff
@@ -1283,7 +1304,7 @@ def _generate_beam_search(
             assert (len(hypo) == max_length for hypo in best)
             decoded = torch.stack(best).type(torch.long).to(next(self.parameters()).device)
 
-        if self.config.is_encoder_decoder:
+        if self.config.is_encoder_decoder and decoder_start_token_id != bos_token_id:
```

Comment:
@thomwolf @sshleifer This if statement might be debatable:
- In my opinion it is a special behavior (of Bart) to cut the first decoded output id and it should only happen if a special `decoder_start_token_id` is set (which happens for Bart).

Reply:
I agree with your opinion: if you force a prefix token at the beginning, it should not be part of the output.

Reply:
See my comments above

Reply:
Actually, removed this if statement completely. 

Think this is the best solution, because:

When decoding bart output with `remove_special_characters=True` as is done in all 
tests and examples, the final decoded solution is the same as before.